README.md

```diff
@@ -45,3 +45,81 @@ defmodule YourPlugBasedThing.Endpoint do
   ...
 end
 ```
+
+## Handling Non-standard Path Parameters
+
+Sometimes we have paths that have path parameters that are not easily
+matchable using regular expressions.
+
+A simple example would be this:
+`/api/v1/status_service/slug/my-custom-slug-4`
+
+A more complex example might look like this:
+`/api/v1/status_service/slug/my-custom-slug-4/views/my-user-readable-view-id`
+
+`PlugDatadogStats` will interpret each of these as a separate path and create a tag
+for all of them which is Not Desirable™.
+
+### Some Potential Approaches
+
+#### Anonymous function
+
+This requires the most work from the user of the plug.  However I think this should likely
+be the approach we take until we hear that many people are running in to this issue.
+
+```elixir
+# The function should return a list in the shape of
+# conn.path_info but with fields removed for dd tags
+# Example: ["api", "v1", "status_service", "slug", "OBFUSCATED"]
+
+pre_plug PlugDatadogStats, path_normalize_fn: fn
+  ["api", "v1", "status_service", "slug", slug_one, "views", slug_two] = path ->
+    path
+    |> Enum.map(fn 
+      ^slug_one ->
+        "SLUG"
+
+      ^slug_two ->
+        "OTHER_SLUG"
+
+      segment ->
+        segment
+      end
+    )
```

Comment:
Just a mechanical comment on this example: I would advise against using `Enum.map` for a static list such as this. It would be a huge performance hit (whether or not you notice it is another matter). Consider instead:
```elixir
fn
  ["api", "v1", "status_service", "slug", _slug_one, "views", _slug_two] ->
    ["api", "v1", "status_service", "slug", "SLUG", "views", "OTHER_SLUG"]
  ["api", "v1", some_service, "slug", _slug_one, "views", _slug_two] ->
    ["api", "v1", some_service, "slug", "SLUG", "views", "OTHER_SLUG"]
  path -> path
end
```
We're just using pattern matching to match a normalization body…and if we're normalizing it then in this case it's a static list! No garbage! No thinking! :) I also included an example of "what if there are other services with a similar structure?". It's still pretty efficient to create a list like this…I'd even hope the compiler keeps the static tail lying around. :)

Reply:
Thanks for pointing this out!  Not sure what I was thinking there.  Looking at your example I think I like this best as a path forward.  It's uncomplicated and powerful enough to handle any case you throw at it.